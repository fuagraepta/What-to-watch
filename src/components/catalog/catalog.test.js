import React from 'react';
import renderer from 'react-test-renderer';

import Catalog from './catalog.jsx';

describe(`The component is rendered correctly`, () => {
  it(`Catalog correctly renders with transferrd mock data`, () => {
    const mockMovies = [
      {
        id: `d90f:fc74:a512:6497:6aab:537:251c:c35a`,
        genre: `Comedies`,
        title: `Johnny English`,
        image: `img/johnny-english.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
      },
      {
        id: `94b2:9524:2d16:7525:dcb8:9540:7ba0:3afc`,
        genre: `Crime`,
        title: `Snatch`,
        image: `img/snatch.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
      }
    ];
    const mockGenres = [`All genres`, `Crime`, `Comedies`];
    const mockCurrentGenre = `All genres`;

    const catalog = renderer
    .create(<Catalog
      movies = {mockMovies}
      allGenres = {mockGenres}
      moviesPerPage = {8}
      currentGenre = {mockCurrentGenre}
      onGenreClick = {jest.fn}
      onShowMoreButtonClick = {jest.fn}
    />)
    .toJSON();

    expect(catalog).toMatchSnapshot();
  });
});
