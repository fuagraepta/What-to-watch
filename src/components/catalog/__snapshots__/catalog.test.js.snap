// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Catalog correctly renders with transferrd mock data 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Comedies
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
        onMouseLeave={[Function]}
      >
        <video
          style={
            Object {
              "height": "100%",
              "objectFit": "fill",
              "width": "100%",
            }
          }
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/films#d90f:fc74:a512:6497:6aab:537:251c:c35a"
        >
          Johnny English
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
        onMouseLeave={[Function]}
      >
        <video
          style={
            Object {
              "height": "100%",
              "objectFit": "fill",
              "width": "100%",
            }
          }
        />
      </div>
      <h3
        className="small-movie-card__title"
        onClick={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/films#94b2:9524:2d16:7525:dcb8:9540:7ba0:3afc"
        >
          Snatch
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button visually-hidden"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
